staging/src/k8s.io/apiserver/plugin/pkg/audit/webhook/webhook.go

```diff
@@ -229,15 +291,43 @@ L:
 		}
 	}
 
+	return events
+}
+
+// collectLastEvents assumes that the buffer was closed. It collects the first
+// maxBatchSize events from the closed buffer into a batch and returns them.
+func (b *batchBackend) collectLastEvents() []auditinternal.Event {
```

Comment:
I am usually a fan of making it explicit if state is not used, and of functional style in general. Here though the complexity is very local, i.e. this is a private method. We don't win much by detaching it wrong `b`. I would leave it the way it is, while the alternative with a detached func would be fine as well.